src/hardware/mouse/mouse.cpp

```diff
@@ -420,8 +420,15 @@ void MOUSE_NewScreenParams(const uint32_t clip_x, const uint32_t clip_y,
 	assert(res_x <= INT32_MAX);
 	assert(res_y <= INT32_MAX);
 
-	state.clip_x = clip_x;
-	state.clip_y = clip_y;
+	if (clip_x * 2 >= res_x || clip_y * 2 >= res_y) {
+		LOG_WARNING("MOUSE: Incorrect screen params: resolution %dx%d, clipping %dx%d",
+		            res_x, res_y, clip_x, clip_y);
+		state.clip_x = 0;
+		state.clip_y = 0;
```

Comment:
The problem is, that we need not "acceptable" values - but values which will allow us to properly calculate mouse cursor position for the guest game/program, or else the seamless mouse integration won't work even with compatible software.

Another problem: I have no idea yet, which strange values might arrive (and they are OK), and which shouldn't, ever.  There are no comments explaining this in our SDL integration code, there code itself is long and complicated, there are many combinations of settings possible, different OS'es (and, in case of Linux, different window managers) might behave differently.

If we have a situation like here (the size of left/top black border is half of the window size or more), than I'm 99% sure there is a bug somewhere, and having a warning might draw the user attention to report the problem.